### PR TITLE
Loads links to related scholarships in college page.

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**
image

### CODE REVIEW

File: src/main/webapp/js/datahandlers/collegepagedatahandler.js [hunk 1]
```diff
@@ -87,6 +89,23 @@ class CollegeDataHandler extends SinglePageDataHandler {
...
+      return scholarshipJson.map((e) => this.formatScholarshipListButton(e));
+    } catch(e) {
+      throw new Error(`Cannot get scholarship json ${e}`);
+    }
+  }
+
+  formatScholarshipListButton(elem) {
+    return {
+      'name' : elem[SCHOLARSHIP_NAME],
+      'id' : elem['id'],
```
Comment (Lorraine-Lyu) [inline src/main/webapp/js/datahandlers/collegepagedatahandler.js:110]: The rendered template shows undefined if I change the strings here to constant variables. I'm not sure how to fix it.

File: src/main/webapp/js/finscholar/collegepageview/collegepageview.js [hunk 2]
```diff
@@ -18,12 +18,41 @@ goog.module('finscholar.collegepageview');
...
+    /** @private {?Element} The container for scholarship buttons. */
+    this.scholarshipContainer_ = null;
+  }
+
+  /** @override */
+  async renderView() {
+    try {
+      await super.renderView();
+      this.scholarshipContainer_ = googDom.getElement(SCHOLARSHIP_DIV_NAME);
+      await this.renderScholarships_();
```
Comment (Lorraine-Lyu) [inline src/main/webapp/js/finscholar/collegepageview/collegepageview.js:40]: We don't need a separate call to add links to colleges in scholarship page, so I did not change structure for singlepageview

File: src/main/webapp/js/datahandlers/collegepagedatahandler.js [hunk 3]
```diff
@@ -87,6 +89,23 @@ class CollegeDataHandler extends SinglePageDataHandler {
       cumulativeMedianDebt: element[MEDIAN_DEBT].toString()
     };
   }
+
+  async findScholarships(id) {
```
Comment (CarlosM10011) [inline src/main/webapp/js/datahandlers/collegepagedatahandler.js:86]: add jsdoc please.

File: src/main/webapp/js/datahandlers/collegepagedatahandler.js [hunk 4]
```diff
@@ -87,6 +89,23 @@ class CollegeDataHandler extends SinglePageDataHandler {
...
+    try {
+      let scholarshipResponse = await fetch(`${FIND_SCHOLARSHIP_ENDPOINT}?id=${id}`);
+      let scholarshipJson = await scholarshipResponse.json();
+      return scholarshipJson.map((e) => this.formatScholarshipListButton(e));
+    } catch(e) {
+      throw new Error(`Cannot get scholarship json ${e}`);
+    }
+  }
+
+  formatScholarshipListButton(elem) {
```
Comment (CarlosM10011) [inline src/main/webapp/js/datahandlers/collegepagedatahandler.js]: add jsdoc please.

File: src/main/webapp/js/finscholar/collegepageview/collegepageview.js [hunk 5]
```diff
@@ -18,12 +18,41 @@ goog.module('finscholar.collegepageview');
...
+    } catch(e) {
+      throw new Error(`Error rendering views for colleges ${e}`);
+    }
+  }
+ 
+  /** Rander buttons for each scholarship related to current college. */
+  async renderScholarships_() {
```
Comment (CarlosM10011) [inline src/main/webapp/js/finscholar/collegepageview/collegepageview.js:47]: Is this meant to be private?

File: src/main/webapp/js/finscholar/singlepageview/singlepageview.js [hunk 6]
```diff
@@ -29,26 +32,76 @@ class SinglePageView extends BasicView {
...
+    this.dataHandler = dataHandler;
 
     /**
      * @private @const {!function(Object): goog.soy.data.SanitizedHtml}
      */
     this.template_ = template;
 
-    /** @private {string} The id of the scholarhsip/college. */
-    this.id_ = '';
+    /** @protected {string} The id of the scholarhsip/college. */
```
Comment (CarlosM10011) [inline src/main/webapp/js/finscholar/singlepageview/singlepageview.js]: typo.

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
Some linter messages:

```
src/main/webapp/js/datahandlers/collegepagedatahandler.js:104: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
  104|     return {
           ^^^^^^^^
  105|       'name' : elem[SCHOLARSHIP_NAME],
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  106|       'id' : elem['id'],
       ^^^^^^^^^^^^^^^^^^^^^^^^
  107|     }
       ^^^^^

src/main/webapp/js/finscholar/collegepageview/collegepageview.js:19: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const googDom = goog.require('goog.dom');
const {CollegeDataHandler} = goog.require('datahandlers.collegepage');
const {SinglePageView} = goog.require('finscholar.singlepageview');
const {collegepage, scholarshiplist} = goog.require('finscholar.collegepageview.templates');

  19| const {CollegeDataHandler} = goog.require('datahandlers.collegepage');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/collegepageview/collegepageview.js:49: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] CollegeDataHandler is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  49|       const nameAndIdList = await (/** @type {CollegeDataHandler} */(this.dataHandler))
                                                    ^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/singlepageview/singlepageview.js:19: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const JsactionActionFlow = goog.require('jsaction.ActionFlow');
const JsactionDispatcher = goog.require('jsaction.Dispatcher');
const JsactionEventContract = goog.require('jsaction.EventContract');
const {BasicView} = goog.require('basicview');
const {SinglePageDataHandler} = goog.require('datahandlers.singlepagedatahandler');

  19| const {BasicView} = goog.require('basicview');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/main/webapp/js/finscholar/singlepageview/singlepageview.js:41: WARNING - [JSC_REDUNDANT_NULLABILITY_MODIFIER_JSDOC] Function is a non-reference type which is already non-nullable.
Please remove the redundant !, which is disallowed by the style guide.
  41|      * @private @const {!function(Object): goog.soy.data.SanitizedHtml}
                              ^

src/main/webapp/js/finscholar/singlepageview/singlepageview.js:41: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] Object is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  41|      * @private @const {!function(Object): goog.soy.data.SanitizedHtml}
                                        ^^^^^^

src/main/webapp/js/finscholar/singlepageview/singlepageview.js:41: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] goog.soy.data.SanitizedHtml is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  41|      * @private @const {!function(Object): goog.soy.data.SanitizedHtml}
                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Review by tinagrasso — COMMENTED** [review]
Can you provide a screenshot of what this looks like?

**Review by hzimz — APPROVED** [review]

**Review by MikolJiron — APPROVED** [review]

File: src/main/webapp/js/datahandlers/collegepagedatahandler.js (hunk 1 above)
Comment (tinagrasso) [reply src/main/webapp/js/datahandlers/collegepagedatahandler.js:110]: please debug and find out.

**Review by tinagrasso — APPROVED** [review]

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
Some linter messages:
```
src/main/webapp/js/datahandlers/collegepagedatahandler.js:82: WARNING - [JSC_MISSING_RETURN_JSDOC] Function with non-trivial return must have JSDoc indicating the return type.
  82|   async findScholarships(id) {
              ^^^^^^^^^^^^^^^^^^^^^^
  83|     try {
      ^^^^^^^^^
...
  89|     }
      ^^^^^
  90|   }
      ^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:94: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] Object is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  94|    * @param {Object} elem The scholarship element to be formatted.
                   ^^^^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:96: WARNING - [JSC_MISSING_RETURN_JSDOC] Function with non-trivial return must have JSDoc indicating the return type.
   96|   formatScholarshipListButton_(elem) {
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   97|     return {
       ^^^^^^^^^^^^
...
  100|     }
       ^^^^^
  101|   };
       ^^^

src/main/webapp/js/datahandlers/collegepagedatahandler.js:97: WARNING - [JSC_MISSING_SEMICOLON] Missing semicolon
   97|     return {
           ^^^^^^^^
   98|       'name' : elem[SCHOLARSHIP_NAME],
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   99|       'id' : elem['id'],
       ^^^^^^^^^^^^^^^^^^^^^^^^
  100|     }
       ^^^^^

src/main/webapp/js/finscholar/singlepageview/singlepageview.js:19: WARNING - [JSC_REQUIRES_NOT_SORTED] goog.require() and goog.requireType() statements are not sorted. The correct order is:

const JsactionActionFlow = goog.require('jsaction.ActionFlow');
const JsactionDispatcher = goog.require('jsaction.Dispatcher');
const JsactionEventContract = goog.require('jsaction.EventContract');
const {BasicView} = goog.require('basicview');
const {SinglePageDataHandler} = goog.require('datahandlers.singlepagedatahandler');

  19| const {BasicView} = goog.require('basicview');
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Also compiling this results in a warning for me.

**Review by CarlosM10011 — CHANGES_REQUESTED** [review]
```
src/main/webapp/js/datahandlers/collegepagedatahandler.js:80: WARNING - [JSC_MISSING_NULLABILITY_MODIFIER_JSDOC] Promise is a reference type with no nullability modifier, which is disallowed by the style guide.
Please add a ! to make it explicitly non-nullable, or a ? to make it explicitly nullable.
  80|    * @return {Promise<!Array<{
                    ^^^^^^^
```

I'm also getting a compiler warning when building the app.

**Review by CarlosM10011 — APPROVED** [review]

**Pull request merged**